source/src/protocols/ligand_docking/Transform.cc

```diff
@@ -612,12 +619,9 @@ bool Transform::check_rmsd(core::conformation::UltraLightResidue const & start,
 {
 	debug_assert(start.natoms() == current.natoms());
 
-	core::Real total_distance =0.0;
-	for ( core::Size atomno = 1; atomno <= start.natoms(); ++atomno ) {
-		total_distance += start[atomno].distance(current[atomno]);
-	}
+	core::Real rmsd = core::scoring::automorphic_rmsd( *start.residue(), *current.residue(), /*superimpose=*/false );
```

Comment:
Part of the purpose of UltraLightResidue is to not have to update the coordinates of the underlying conformation::Residue and take the overhead of that. As such, the `residue()` coordinates are set at initialization and not updated -- your change here isn't doing what you want it to (it's using stale coordinates).